aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttManager.java

```diff
@@ -210,6 +213,30 @@ public boolean isMetricsEnabled() {
     /** Override value for System.currentTimeInMillis.  Used for unit testing reconnect logic. */
     private Long unitTestMillisOverride;
 
+    /**
+     * Sets the userMetaData map.
+     *
+     * @param userMetaDataMap userMetaData map
+     */
+    public void addUserMetaData(Map<String, String> userMetaDataMap) {
+        StringBuilder userMetadata = new StringBuilder("?SDK=Android&Version=" + SDK_VERSION);
+
+        if (userMetaDataMap != null) {
+            // Append each of the user-specified key-value pair to the user metadata for the connection
+            for (Map.Entry<String, String> metaData : userMetaDataMap.entrySet()) {
+                userMetadata.append("&" + metaData.getKey() + "=" + metaData.getValue());
+            }
+        }
+
+        if(userMetadata.length() > 255) {
+            LOGGER.warn("Too many characters. User metadata was truncated.", new IllegalArgumentException("Total number of characters in user metadata" +
+                    " cannot exceed " + (255 - ("?SDK=Android&Version=" + SDK_VERSION).length())));
```

Comment:
Aovid the code duplication here by capturing the length after line 222 into a local variable

Reply:
Done